packages/theming/atlas/src/themesource/atlas_ui_resources/web/core/helpers/_badge.scss

```diff
@@ -6,43 +6,29 @@
 //
 
 /* ==========================================================================
-   Labels
+   Badge-web
 
 //== Design Properties
 //## Helper classes to change the look and feel of the component
 ========================================================================== */
-// Color variations
-.label-default {
-    color: $label-default-color;
-    background-color: $label-default-bg;
-}
+// Style variations
 
-.label-primary {
-    color: $label-primary-color;
-    background-color: $label-primary-bg;
+.widget-badge.label-secondary {
```

Comment:
It might break current projects using this class

Reply:
Hasn't `widget-badge` always existed in the widget's top-level classnames? If so, it shouldn't break imo. What do you think @IIsaku ?

Reply:
If users were adding the class to something else to give same appearance now we are blocking then adding widget-badge as parent

Reply:
- `widget-badge` was in the previous version also applied to the badge-web widget.
- In _lables.scss (packages/theming/atlas/src/themesource/atlas_ui_resources/web/core/widgets/_labels.scss) we are still shipping the label styles which is used when applied to other widgets.
- If users defined a custom label class (like `.label-success`) in the custom theme file and applied it to badge, it will always be overruled by the default badge style. Users will have to change their custom class (like `.widget-badge.label-success`). So yes, it can break.

Reply:
So why?

Reply:
We need to apply a more specific `.widget-badge.label-success` css rule to overrule the default `.widget-badge` style  that is applied to the same `span` element. This default style contains the primary brand style in case no `Style` design property is configured. Before, style overruling was relying on the order of css rules in files, which is very fragile when Atlas changes will be made in the future.

Reply:
@cjfhodges @wesleysieses what do you guys think about this breaking change in the styles?

Reply:
I didn't think we had to be this prescriptive for the class because in main.scss we list /core/badge.scss first then the /helper/badge.scss. So the helper styles should override the badge ones where necessary.

Reply:
I do agree that the order of imports is fickle and prefer strengthening selectors where needed.

The only outstanding issue here is the breaking change for end-users where custom styles have been added with e.g. `.label-success` - they will be overpowered. Can we discuss in standup? I'm struggling to find the `_custom.scss` file in the monorepo.

Reply:
See this thread [here](https://mendix.slack.com/archives/GV93BNVDM/p1613635194001900). Until we fully understand how end-users achieve custom styling nowadays, this issue will be difficult to solve.

Reply:
Fixed as discussed in standup